OutOfSchool/OutOfSchool.AuthCommon/Validators/CustomPasswordValidator.cs

```diff
@@ -0,0 +1,28 @@
+﻿using static OutOfSchool.AuthCommon.Validators.GeneratedRegexes;
+
+namespace OutOfSchool.AuthCommon.Validators;
+
+public class CustomPasswordValidator(string passwordValidationErrorMessage, string passwordRequiredErrorMessage) : IPasswordValidator<User>
+{
+    public Task<IdentityResult> ValidateAsync(UserManager<User> manager, User user, string? password)
+    {
+        List<IdentityError> errors = [];
```

Comment:
(Logic, performance) Please let me understand why we need to have list of errors in the code that could have max 1 error? You create an empty list, then add something there (thus "reallocating" the list), then convert it to array using spread operator - excessive CPU, code, memory pressure. Why? 
Please consider usage of simple immediate `return` in each `if` for errors, and at method end for no errors.